src/app/dataset-view/additional-components/flows-component/components/flows-table/flows-table.component.ts

```diff
@@ -137,22 +187,95 @@ export class FlowsTableComponent implements OnInit, OnChanges {
                 noButtonText: "Cancel",
                 handler: (ok) => {
                     if (ok) {
-                        this.cancelFlowChange.emit(flowId);
+                        this.cancelFlowChange.emit({ flowId, datasetId });
                     }
                 },
             }),
         );
     }
 
-    public navigateToFlowDetaisView(flowId: string): void {
+    public navigateToFlowDetaisView(flow: FlowSummaryDataFragment, datasetId: string): void {
         this.navigationService.navigateToFlowDetails({
-            accountName: this.datasetBasics.owner.accountName,
-            datasetName: this.datasetBasics.name,
-            flowId,
+            accountName: this.accountView && this.accountName ? this.accountName : this.datasetBasics.owner.accountName,
+            datasetName: this.accountView ? this.datasetByIdForAccount(datasetId).name : this.datasetBasics.name,
+            flowId: flow.flowId,
         });
     }
 
     public dynamicImgSrc(status: FlowStatus): string {
         return DatasetFlowDetailsHelpers.dynamicImgSrc(status);
     }
+
+    public accountFormatter(x: Account | string): string {
+        return typeof x !== "string" ? x.accountName : x;
+    }
+
+    public searchByAccountName: OperatorFunction<string, readonly Account[]> = (text$: Observable<string>) =>
+        text$.pipe(
+            debounceTime(200),
+            distinctUntilChanged(),
+            tap(() => (this.searchingAccount = true)),
+            switchMap((term) =>
+                this.flowService.flowsInitiators(this.datasetBasics.id).pipe(
+                    map((owners: Account[]) =>
+                        owners
+                            ?.filter((owner: Account) => owner.accountName.toLowerCase().indexOf(term) > -1)
+                            .slice(0, 10)
+                            .sort((a, b) => {
+                                if (a.accountName < b.accountName) return -1;
+                                if (a.accountName > b.accountName) return 1;
+                                return 0;
+                            }),
+                    ),
+                    catchError(() => {
+                        return of([]);
+                    }),
+                ),
+            ),
+            tap(() => (this.searchingAccount = false)),
+        );
+
+    public searchByDatasetName: OperatorFunction<string, readonly Dataset[]> = (text$: Observable<string>) =>
+        text$.pipe(
+            debounceTime(200),
+            distinctUntilChanged(),
+            tap(() => (this.searchingDataset = true)),
+            switchMap((term) =>
+                this.accountService.getDatasetsWithFlows(this.accountName ?? "").pipe(
```

Comment:
Same problem here. Re-querying list of datasets on every character!
This data does not change until we refresh the page.

Reply:
Fixed